### ISSUE TITLE
ProgbarLogger no longer prints out custom metric implemented in callbacks

### ISSUE THREAD

**Issue by Guzzii** [keras-team/keras#11522]
Hi guys. I notice one change made in the latest version of keras. The `ProgbarLogger` callback is now created right after `BaseLogger`, as shown here https://github.com/keras-team/keras/blob/master/keras/engine/training_arrays.py#L96-L107. I am wondering what is the reason why the implementation was changed from `ProgbarLogger` being the last callback appended to the callback list.

The previous implementation seems a bit more diverse to me, because you can manipulate the logs before ProgbarLogger prints it out, so that you can implement your own metric in a callback. But with the current implementation, this is not possible, simply due to the fact that ProgbarLogger would print the logs out before hitting custom callbacks.

Any thoughts?

- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
`pip install git+git://github.com/keras-team/keras.git --upgrade --no-deps`

**Comment by gabrieldemarmiesse** [comment]
Can you describe your use case? Metrics should be defined and put in the `compile` call. Why do you use a callback for that?

**Comment by Guzzii** [comment]
Hi @gabrieldemarmiesse. Thanks for your reply. The reason why I implemented the metric in callback is due to complicated metrics from business units. As a simple example, we might be interested in rank correlation within some kind of groups. To compute that, one has to pass `a group index` along  with` y_true` and `y_predict` to the metric function. To the best of my knowledge, it is challenging to do with `compile` call. However it is pretty straightforward with `on_epoch_end` in callbacks. I would be happy to learn if it can be done with compile. Please let me know if this makes sense.

**Comment by gabrieldemarmiesse** [comment]
So you use numpy to compute your metric, right?

**Comment by Guzzii** [comment]
I think `numpy` can definitely do it. But I have been mainly using pandas as the easiest solution.
